pykdl_ros/src/pykdl_ros/pykdl_ros.py

```diff
@@ -0,0 +1,139 @@
+from rospy import Time
+import PyKDL as kdl
+from std_msgs.msg import Header
+
+
+class VectorStamped:
+    """
+    Stamped version of PyKDL.Vector
+    """
+
+    def __init__(self, vector: kdl.Vector, stamp: Time, frame_id: str):
+        """
+        Constructor
+
+        :param vector: vector
+        :type vector: PykDL.Vector
+        :param stamp: TimeStamp
+        :type stamp: rospy.Time
+        :param frame_id: Frame ID
+        :type frame_id: str
+        """
+        assert isinstance(vector, kdl.Vector)
+        assert isinstance(stamp, Time)
+        assert isinstance(frame_id, str)
+        self.vector = vector
+        self.header = Header(frame_id=frame_id, stamp=stamp)
+
+    def __repr__(self):
+        xyz = "(x={x}, y={y}, z={z})".format(x=self.vector.x(), y=self.vector.y(), z=self.vector.z())
+        return "VectorStamped({xyz} @ {fid})".format(xyz=self.vector, fid=self.header.frame_id)
+
+    def __eq__(self, other):
+        if isinstance(other, VectorStamped):
```

Comment:
Why do we allow it to be anything else then a VectorStamped?

Reply:
no need to crash the code when comparing apple and a peer.